File: MosPosudit.Model/Requests/Category/CategoryInsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MosPosudit.Model.Requests.Category
{
    public class CategoryInsertRequest
    {
        [Required(ErrorMessage = "Category name is required")]
        [StringLength(100, ErrorMessage = "Category name cannot be longer than 100 characters")]
        public string Name { get; set; } = string.Empty;

        [StringLength(500, ErrorMessage = "Description cannot be longer than 500 characters")]
        public string? Description { get; set; }

        public string? ImageBase64 { get; set; }
    }
}

File: MosPosudit.UI/shared/lib/dtos/category/category_insert_request.dart
class CategoryInsertRequestDto {
  final String name;
  final String? description;
  final String? imageBase64;

  CategoryInsertRequestDto({
    required this.name,
    this.description,
    this.imageBase64,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'description': description,
        'imageBase64': imageBase64,
      };
}

